src/events/superagent-wrapper.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * @fileoverview Wrapping superagent-wrapper's http library, since we can't trace calls
+ * (async_hooks params always equals 0)
+ */
+
+const {
+    tracer,
+    moduleUtils,
+    eventInterface,
+    utils,
+    httpHelpers,
+} = require('epsagon');
+const { EPSAGON_HEADER } = require('../http.js');
+
+/**
+ * Wraps the superagent-wrapper http send command function with tracing
+ * @param {Function} wrappedFunction The wrapped function from superagent-wrapper module
+ * @returns {Function} The wrapped function
+ */
+function superagentWrapper(wrappedFunction) {
+    return function internalSuperagentClientWrapper(req) {
+        const response = wrappedFunction.apply(this, [req]);
+        try {
+            const { hostname, pathname: path } = new URL(req.url);
+
+            const { slsEvent: httpEvent, startTime } =
+                eventInterface.initializeEvent(
+                    'http',
+                    hostname,
+                    req.method,
+                    'http'
+                );
+
+            const epsagonTraceId = httpHelpers.generateEpsagonTraceId();
+            // Inject header to support tracing over HTTP requests
+            if ((process.env.EPSAGON_DISABLE_HTTP_TRACE_ID || '').toUpperCase() !== 'TRUE') {
+                req.header[EPSAGON_HEADER] = epsagonTraceId;
+            }
+
+            eventInterface.addToMetadata(httpEvent,
+                {
+                    url: req.url,
```

Comment:
what with the body of the requests?

Reply:
Unfortunately, it's not reachable from this position (function) - it only comes as an async uncorrelated event afterwards, so we can't match it.